### ISSUE TITLE
Pieces aren't working!

### ISSUE THREAD

**Issue by jonathanpike** [teamendgame/endgame-chess#59]
If I try to move a piece, the AJAX request is returning 404. You can see the console here: 
![screenshot 2015-12-14 15 01 02](https://cloud.githubusercontent.com/assets/8129357/11792369/babcab62-a273-11e5-92f8-ead34f39d232.png)

**Comment by jonathanpike** [comment]
Here's what the console spits out with the error: 

```
Started PUT "/pieces/40" for 10.0.2.2 at 2015-12-14 22:31:31 +0000
Cannot render console from 10.0.2.2! Allowed networks: 127.0.0.1, ::1, 127.0.0.0/127.255.255.255
Processing by PiecesController#update as JSON
  Parameters: {"piece"=>{"row_position"=>"2", "col_position"=>"5"}, "id"=>"40"}
  Game Load (0.3ms)  SELECT  "games".* FROM "games" WHERE "games"."id" = $1 LIMIT 1  [["id", nil]]
Completed 404 Not Found in 2ms (ActiveRecord: 0.3ms)

ActiveRecord::RecordNotFound (Couldn't find Game with 'id'=):
  app/controllers/pieces_controller.rb:45:in `check_player_color'
```

**Comment by jonathanpike** [comment]
The bug was in the `check_player_color` method.  Since it was being called with the `before_action` callback, any time `:game_id` wasn't in params (ie. whenever we're doing something with the pieces controller), it was returning 404.  

I've fixed this and will be pushing up a PR.

**Comment by jonathanpike** [comment]
Solved in #63
